app/models/solr_document.rb

```diff
@@ -584,8 +584,67 @@ def extract_isbn(elements:, tag: "020", sfield: "a", qfield: "q")
   end
 
   def get_related_records
-    related = RelatedRecords.new(self)
-    related.in_collection? ? related : []
+    ids = get_marc_derived_field("0359")
+    collection_id = ""
+    if ids.present?
+      ids.each do |id|
+        if id.match(/\(.*\)/)
+          collection_id = id
+          break
+        end
+      end
+    end
+
+    related_records = []
+
+    # Check for related collections in the 973 subfield
+    related_973_records = []
+    related_973 = get_marc_derived_field("973w")
+    related_973.each do |r|
+      rec = RelatedRecords.new(self, collection_id)
+      rec.subfield = "973"
+      rec.parent_id = r
+      related_973_records << rec if rec.in_collection?
+    end
+
+    # Check for related collections in the 773 subfield
+    related_773_records = []
+    related_773 = get_marc_derived_field("773w")
+    related_773.each do |r|
+      rec = RelatedRecords.new(self, collection_id)
+      rec.subfield = "773"
+      rec.parent_id = r
+      related_773_records << rec if rec.in_collection?
+    end
+
+    # If there are no 973 or 773 records, but there is a collection ID, then this could be a
+    # parent collection.
+    if related_973_records.blank? && related_773_records.blank?
+      related_records << RelatedRecords.new(self, collection_id)
+    else
+      related_records += related_973_records
+      related_records += related_773_records
+    end
+
```

Comment:
Don't know if it matters, but VuFind put the 773 before the 973. Don't know if that's because 773 is core MARC and 973 is custom.

Reply:
It didn't seem to make any difference in my testing, but I'll re-order them just in case.

Reply:
I don't know I've ever come across any records with both, it was more whether any NLA people would notice or care. I just noticed in VuFind there were some specific orderings, I've probably just caught some hyper-awareness disease from looking at VuFind code.